core/src/main/java/org/arquillian/cube/impl/client/container/remote/CubeAuxiliaryArchiveAppender.java

```diff
@@ -12,10 +15,17 @@
 
     @Override
     public Archive<?> createAuxiliaryArchive() {
-        return ShrinkWrap.create(JavaArchive.class, "arquillian-cube.jar")
+        JavaArchive javaArchive = ShrinkWrap.create(JavaArchive.class, "arquillian-cube.jar")
                 .addPackage(CubeController.class.getPackage())
                 .addPackages(true, CubeRemoteExtension.class.getPackage())
                 .addClass(CubeControllerProvider.class)
                 .addAsServiceProvider(RemoteLoadableExtension.class, CubeRemoteExtension.class);
+
+        if (LoadableExtension.Validate.classExists("org.arquillian.cube.requirement.ArquillianConditionalRunner") &&
+                LoadableExtension.Validate.classExists("org.arquillian.cube.spi.requirement.Requires")) {
```

Comment:
but this is not an assumption, if you are using `CubeAuxiliaryArchiveAppender` it means that `spi` is also there because it is a direct dependency. Also keep in mind that checking if a class is on classpath is not a free operation it takes some time to do it so it is something to take into consideration. But of course if needed it should be done, but this is not the case because it is direct.